sys/net/gnrc/link_layer/lwmac/tx_state_machine.c

```diff
@@ -451,9 +452,7 @@ static bool _send_data(gnrc_netif2_t *netif)
     int res = _gnrc_lwmac_transmit(netif, pkt);
     if (res < 0) {
         LOG_ERROR("ERROR: [LWMAC-tx] Send data failed.");
-        if (pkt != NULL) {
```

Comment:
Isn't it sometimes that the `_gnrc_lwmac_transmit` function (which is copied from the `_send` function from `gnrc_netif2_ieee802154.c`) leave the packet unreleased?

Reply:
If `pkt` can be `NULL` the whole function is defective (you access members of `pkt` several times above). Otherwise, this check is redundant. The release still happens (just without a check) below.

Reply:
(I did not really understand what you mean, so if I'm jumping to conclusions, please correct me)

Reply:
Ah, yes, you are right. I see the function is redundant upon error return of the send function.